File: elixir/sublist/lib/sublist.ex
defmodule Sublist do
  @doc """
  Returns whether the first list is a sublist or a superlist of the second list
  and if not whether it is equal or unequal to the second list.
  """

  def compare(a, b) do
    cond do
      equal?(a, b) -> :equal
      sublist?(b, a) -> :sublist
      sublist?(a, b) -> :superlist
      true -> :unequal
    end
  end

  defp equal?(a, b), do: a === b

  defp sublist?(a, b) when length(a) < length(b), do: false
  defp sublist?(a, b) do
    sub = Enum.take(a, length(b))
    equal?(sub, b) or sublist?(tl(a), b)
  end
end

File: clojure/sublist/src/sublist.clj
(ns sublist)

(defn ^:private sublist? [list1 list2]
  (some #{list1} (partition (count list1) 1 list2)))

(defn ^:private superlist? [list1 list2]
  (sublist? list2 list1))

(defn classify [list1 list2] ;; <- arglist goes here
  (cond
    (= list1 list2)        :equal
    (sublist? list1 list2) :sublist
    (superlist? list1 list2) :superlist
    :else :unequal))


